api.js

```diff
@@ -177,24 +187,30 @@ Api.prototype.run = function (files) {
 			var tests = files.map(self._runFile);
 
 			// receive test count from all files and then run the tests
+			var runOnlyExclusive = false;
 			var statsCount = 0;
 
 			return new Promise(function (resolve) {
 				tests.forEach(function (test) {
 					var counted = false;
 
-					function tryRun() {
+					function tryRun(stats) {
 						if (counted) {
 							return;
 						}
 
+						runOnlyExclusive = runOnlyExclusive || stats.hasExclusive;
```

Comment:
It does seem redundant but it's to avoid responsibility creep. `_handleStats()` is concerned with gathering the total `testCount` before tests run while this method is waiting for each test to emit stats before starting them all. Probably the `stats` event itself is overloaded by using it for both purposes.

I'll leave `_handleStats()` as it is but wrap the `tryRun()` on `test.catch()`.